rclpy/src/rclpy/_rclpy.c

```diff
@@ -3914,7 +3915,7 @@ rclpy_create_duration(PyObject * Py_UNUSED(self), PyObject * args)
 
   rcl_duration_t * duration = (rcl_duration_t *) PyMem_Malloc(sizeof(rcl_duration_t));
   if (!duration) {
-    PyErr_Format(PyExc_RuntimeError, "Failed to allocate memory for duration.");
+    PyErr_Format(RCLError, "Failed to allocate memory for duration.");
```

Comment:
This is a memory allocation error:
```suggestion
    PyErr_Format(PyErr_NoMemory, "Failed to allocate memory for duration.");
```